src/pages/AMM/Swap/components/SwapForm/SwapForm.tsx

```diff
@@ -0,0 +1,183 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { ChangeEventHandler, FormEventHandler, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+import { useDebounce } from 'react-use';
+
+import { StoreType } from '@/common/types/util.types';
+import { Card, CardProps, Divider, Flex, H1, TokenInput } from '@/component-library';
+import { AuthCTA } from '@/components/AuthCTA';
+import { SwapPair, SwapSlippage } from '@/types/swap';
+import { useGetCurrencies } from '@/utils/hooks/api/use-get-currencies';
+
+import { useSwapFormData } from '../../hooks/use-swap-form-data';
+import { SlippageManager } from '../SlippageManager';
+import { SwapInfo } from '../SwapInfo';
+import { SwapDivider } from './SwapDivider';
+
+// TODO: here we will call the lib to get the expected output
+const getOutput = async (param?: number) => {
+  return param;
+};
+
+const getPairChange = (pair: SwapPair, currency: CurrencyExt, name: string): SwapPair => {
+  switch (name) {
+    case FormFields.INPUT_TICKER: {
+      if (currency.ticker === pair.output?.ticker) {
```

Comment:
Since we only have one condition, I think a ternary might be better than an if statement.